lightning/src/routing/router.rs

```diff
@@ -149,9 +161,31 @@ impl cmp::PartialOrd for RouteGraphNode {
 struct DummyDirectionalChannelInfo {
 	cltv_expiry_delta: u32,
 	htlc_minimum_msat: u64,
+	htlc_maximum_msat: Option<u64>,
 	fees: RoutingFees,
 }
 
+#[derive(Clone)]
+struct RouteHopInGraph {
+	route_hop: RouteHop,
+	available_msat: Option<u64>,
+	/// Minimal fees required to route to the source node of this hop (one hop "before" using this hop).
+	/// Useful to compute the cost of reaching the source of this hop, which can be later used to choose
+	/// the cheapest, while we traverse backwards.
```

Comment:
I don't really get "source node of this hop".  You have Alice sending a HTLC to Caroll through Bob relay, who is whom ?

Also inbound is confusing as we use it at the channel-level to designate either HTLC direction, it is the same here ?